### PR TITLE
feat(provider/oraclebmcs): Implement bake stage pipeline

### PR DESCRIPTION
**Pull request by owainlewis**
Implement BMCS bake stage pipeline step

### CODE REVIEW

File: app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        <input type="radio"
+               ng-model="stage.region"
+               ng-value="region"
+               ng-change="constrainNetworkOptions(); constrainAvailabilityDomainOptions(); constrainSubnetOptions()"/>
+        {{region}}
+      </label>
+    </stage-config-field>
+
+    <stage-config-field label="Network", help-key="oraclebmcs.pipeline.config.bake.network">
+      <select id="id"
```
Comment (danielpeach) [inline app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.html]: Please drop the `id` attribute here and below.

File: app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      provides: 'bake',
+      cloudProvider: 'oraclebmcs',
+      label: 'Bake',
+      description: 'Bakes an image in the specified region',
+      templateUrl: require('./bakeStage.html'),
+      executionDetailsUrl: require('./bakeExecutionDetails.html'),
+      executionLabelTemplateUrl: require('core/pipeline/config/stages/bake/BakeExecutionLabel'),
+      defaultTimeoutMs: 60 * 60 * 1000, // 60 minutes
+      validators: [
+        {type: 'requiredField', fieldName: 'baseOs', preventSave: true},
```
Comment (danielpeach) [inline app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.js]: Using `preventSave` here is inconsistent with the other providers, but that's up to you.

File: app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        if (!$scope.account && $scope.application.defaultCredentials.oraclebmcs) {
+          $scope.account = $scope.application.defaultCredentials.oraclebmcs;
+        }
+        if (results.baseOsOptions.baseImages.length > 0) {
+          $scope.baseOsOptions = results.baseOsOptions;
+        }
+        if (results.regions.length > 0) {
+          $scope.regionOptions = results.regions;
+        }
+        if (results.availabilityDomains && results.availabilityDomains != {}) {
```
Comment (danielpeach) [inline app/scripts/modules/oracle/pipeline/stages/bake/bakeStage.js]: `{} !== {}` is always true in JS. I would use `Object.keys(results.availabilityDomains).length > 0`.

**Comment by owainlewis** [comment]
Thanks @danielpeach updated based on comments. Rebased also.

File: app/scripts/modules/oracle/pipeline/stages/bake/bakeExecutionDetails.controller.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+} from '@spinnaker/core';
+
+const angular = require('angular');
+
+module.exports = angular.module('spinnaker.oraclebmcs.pipeline.stage.bake.executionDetails.controller', [
+  EXECUTION_DETAILS_SECTION_SERVICE,
+  require('angular-ui-router').default,
+  require('core/delivery/details/executionDetailsSectionNav.directive.js'),
+])
+  .controller('oraclebmcsBakeExecutionDetailsCtrl', function ($log, $scope, $stateParams, executionDetailsSectionService, $interpolate, settings) {
```
Comment (owainlewis) [inline app/scripts/modules/oracle/pipeline/stages/bake/bakeExecutionDetails.controller.js]: import { SETTINGS } from '@spinnaker/core';

File: app/scripts/modules/oracle/pipeline/stages/bake/bakeExecutionDetails.controller.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  EXECUTION_DETAILS_SECTION_SERVICE,
+  SETTINGS
+} from '@spinnaker/core';
+
+const angular = require('angular');
+
+module.exports = angular.module('spinnaker.oraclebmcs.pipeline.stage.bake.executionDetails.controller', [
+  EXECUTION_DETAILS_SECTION_SERVICE,
+  require('angular-ui-router').default,
+  require('core/delivery/details/executionDetailsSectionNav.directive.js'),
```
Comment (anotherchrisberry) [inline app/scripts/modules/oracle/pipeline/stages/bake/bakeExecutionDetails.controller.js]: remove this line

**Pull request merged**